modal/_utils/function_utils.py

```diff
@@ -190,6 +190,33 @@ def serialized_function(self) -> bytes:
             logger.debug(f"Serializing function for class service function {self.cls.__qualname__} as empty")
             return b""
 
+    def get_cls_vars(self) -> Dict[str, Any]:
+        if self.cls is not None:
+            cls_vars = {
+                attr: getattr(self.cls, attr)
+                for attr in dir(self.cls)
+                if not callable(getattr(self.cls, attr)) and not attr.startswith("__")
+            }
+            return cls_vars
+        return {}
+
+    def get_cls_var_attrs(self) -> Dict[str, Any]:
+        import dis
+
+        import opcode
+
+        LOAD_ATTR = opcode.opmap["LOAD_ATTR"]
+        STORE_ATTR = opcode.opmap["STORE_ATTR"]
```

Comment:
Is it weird to rebuild if users only _store_ a new value for a class attribute, but don't read it, during the build phase? (Actually, separate discussion, but since we're doing this anyay we should maybe warn or error if users try to set class/instance attributes in `@build`, since that won't propagate to runtime containers, and they get confused about it occasionally.)

Reply:
Oh this is fair